frontend/components/bc/BcInputSearch.vue

```diff
@@ -0,0 +1,322 @@
+<script setup lang="ts" generic="T">
+const props = withDefaults(defineProps<{
+  hasError?: boolean,
+  hasFocus?: boolean,
+  isLoading: boolean,
+  results: T[] | undefined,
+  shouldClearOnSubmit?: boolean,
+  shouldSelectFirstResult?: boolean,
+}>(), {
+  shouldClearOnSubmit: true,
+})
+
+const emit = defineEmits<{
+  (e: 'search', value: string): void,
```

Comment:
np: I wonder if this works and make things clearer 🤔 
```suggestion
  (e: 'search', input: string): void,
```

Reply:
a little bit yeah...lets do it